artemis-journal/src/main/java/org/apache/activemq/artemis/core/journal/impl/JournalImpl.java

```diff
@@ -1161,20 +1162,27 @@ public void run() {
 
    @Override
    public void appendDeleteRecord(final long id, final boolean sync, final IOCompletion callback) throws Exception {
-
       if (logger.isTraceEnabled()) {
          logger.trace("scheduling appendDeleteRecord::id=" + id);
       }
-
-
       checkJournalIsLoaded();
       lineUpContext(callback);
-      SimpleFuture<Boolean> future = new SimpleFutureImpl<>();
-      internalAppendDeleteRecord(id, sync, (t, v) -> future.set(v), callback);
-      if (!future.get()) {
-         throw new IllegalStateException("Cannot find add info " + id);
+      final SimpleFuture<Boolean> onFoundAddInfo;
+      if (!sync && (callback == null || callback == DummyCallback.getInstance())) {
+         onFoundAddInfo = null;
+      } else {
+         onFoundAddInfo = new SimpleFutureImpl<>();
+      }
+      if (onFoundAddInfo == null) {
+         internalAppendDeleteRecord(id, false, null, callback);
+      } else {
+         internalAppendDeleteRecord(id, sync, (record, result) -> onFoundAddInfo.set(result), callback);
+      }
+      if (onFoundAddInfo != null) {
+         if (!onFoundAddInfo.get()) {
```

Comment:
I'm thinking we should just stop doing that.. just always log, like we do on updates...

The only outcome from the IllegalStateException would be log.warn in most cases.. especially on delete.